Frontend/ui-driver/src/Flow.purs

```diff
@@ -1255,16 +1262,23 @@ bookingOptionsFlow :: FlowBT String Unit
 bookingOptionsFlow = do
   action <- UI.bookingOptions
   case action of
-    SELECT_CAB state -> do
+    SELECT_CAB state toggleDowngrade -> do
+      void $ lift $ lift $ loaderText (getString LOADING) (getString PLEASE_WAIT_WHILE_IN_PROGRESS)
+      void $ lift $ lift $ toggleLoader true
       let canDowngradeToSedan = isJust $ (filter (\item -> item.vehicleVariant == "SEDAN" && item.isSelected) state.data.downgradeOptions) !! 0
           canDowngradeToHatchback = isJust $ (filter (\item -> item.vehicleVariant == "HATCHBACK" && item.isSelected) state.data.downgradeOptions) !! 0
           canDowngradeToTaxi = isJust $ (filter (\item -> item.vehicleVariant == "TAXI" && item.isSelected) state.data.downgradeOptions) !! 0
       let (UpdateDriverInfoReq initialData) = mkUpdateDriverInfoReq ""
           requiredData = initialData{canDowngradeToSedan = Just canDowngradeToSedan,canDowngradeToHatchback = Just canDowngradeToHatchback,canDowngradeToTaxi = Just canDowngradeToTaxi}
       (UpdateDriverInfoResp updateDriverResp) <- Remote.updateDriverInfoBT ((UpdateDriverInfoReq) requiredData)
-      modifyScreenState $ BookingOptionsScreenType (\bookingOptions -> BookingOptionsScreenData.initData{data{ downgradeOptions = state.data.downgradeOptions }})
       modifyScreenState $ DriverProfileScreenStateType (\driverProfile -> driverProfile{ data{ vehicleSelected = transformSelectedVehicles state.data.downgradeOptions} })
-      driverProfileFlow
+      if toggleDowngrade then do
+        modifyScreenState $ BookingOptionsScreenType (\bookingOptions -> bookingOptions{ props{ downgraded = not state.props.downgraded } })
+        void $ lift $ lift $ toggleLoader false
```

Comment:
put `void $ lift $ lift $ toggleLoader false` outside of the if else block
No need to specify at two places